### PR TITLE
1. Unit tests

### PR DESCRIPTION
**Pull request by diarmaidlindsay**
### Issue

- N/A

### Motivation and Context

- Write Sample Unit Tests which demonstrate mocking of the Repository and API Client.

### Modified points

- Added Unit Tests.

### Description and Screenshots

image

### Figma or screen image URL

- N/A

### Worries or Issues

- N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -65,6 +65,7 @@ unlinked.ds
 unlinked_spec.ds
 **/*.g.dart
 **/*.freezed.dart
+**/*.mocks.dart
```
Comment (diarmaidlindsay) [inline .gitignore]: Ignore generated mocks

File: lib/data/repository/my_anime_list_repository.dart [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
-
+import '../api/api_client.dart';
 import '../model/api/request/get_top_anime_request.dart';
 import '../model/api/response/get_top_anime_response.dart';
-import '../providers/api_client_provider.dart';
 
 class MyAnimeListRepository {
   Future<GetTopAnimeResponse> getTopAnime({
-    required Ref ref,
+    required ApiClient apiClient,
```
Comment (diarmaidlindsay) [inline lib/data/repository/my_anime_list_repository.dart]: Make the repository mockable by removing dependency on Riverpods.

File: test/presentation/feature/detail/detail_page_test.dart [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
 
+final _mockMyAnimeListRepository = MockMyAnimeListRepository();
+
 Future<void> _loadTopAnimePage(WidgetTester tester) async {
   await tester.pumpWidget(
     ProviderScope(
       overrides: [
         myAnimeListRepositoryProvider
-            .overrideWithValue(MyAnimeListRepositoryDummy())
+            .overrideWithValue(_mockMyAnimeListRepository)
```
Comment (diarmaidlindsay) [inline test/presentation/feature/detail/detail_page_test.dart]: Use mock instead of dummy class.

**Review by dinkar1708 — COMMENTED** [review]
better to have
1. pr description a litt
2. read me section about document follow for test case
3. how to run test case - command is better if can

**Review by enricmacias — APPROVED** [review]
LGTM

**Comment by enricmacias** [comment]
Code looks good, but as Dinakar says could be good to have a section in the README explaining about the unit tests

**Review by dinkar1708 — APPROVED** [review]
nice
lgtm

**Pull request merged**